### PR TITLE
Missing echo if no desktop exists

### PR DESCRIPTION
**Pull request by aggsol**
See title and change.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,20 +1,15 @@
 # Bing Wallpapers For Linux
```
Comment (whizzzkid) [inline README.md:1]: Any changes to Readme will not be accepted.

File: bingwallpaper [hunk 2]
```diff
@@ -1,5 +1,7 @@
-#!/bin/bash
```
Comment (whizzzkid) [inline bingwallpaper:1]: why do we need to change this?

File: bingwallpaper [hunk 3]
```diff
@@ -36,41 +48,34 @@ while getopts "1" opt; do
...
 
-########################################################################
-#### DO NOT EDIT BELOW THIS LINE #######################################
-########################################################################
-
 # Required Image Uri.
-reqImg=$bing$api$format$day$market$const
-
-while [ 1 ]
+reqImg="$bing$api$format$day$market$const"
```
Comment (whizzzkid) [inline bingwallpaper:58]: not needed

File: bingwallpaper [hunk 4]
```diff
@@ -36,41 +48,34 @@ while getopts "1" opt; do
...
+#echo "request: $reqImg"
```
Comment (whizzzkid) [inline bingwallpaper:59]: remove this comment

File: bingwallpaper [hunk 5]
```diff
@@ -36,41 +48,34 @@ while getopts "1" opt; do
...
+while  true
```
Comment (whizzzkid) [inline bingwallpaper:60]: not needed

File: bingwallpaper [hunk 6]
```diff
@@ -36,41 +48,34 @@ while getopts "1" opt; do
...
 do
 
   # Logging.
   echo "Pinging Bing API..."
 
   # Fetching API response.
-  apiResp=$(curl -s $reqImg)
-  if [ $? -gt 0 ]; then
-    echo "Ping failed!"
+  apiResp=$(curl -s "$reqImg")
```
Comment (whizzzkid) [inline bingwallpaper:67]: ditto

File: bingwallpaper [hunk 7]
```diff
@@ -36,41 +48,34 @@ while getopts "1" opt; do
...
+  if [[ -z "$apiResp" ]]; then
+    error "No valid API response"
     exit 1
   fi
 
-  # Default image URL in case the required is not available.
```
Comment (whizzzkid) [inline bingwallpaper:65]: why was the comment removed?

File: bingwallpaper [hunk 8]
```diff
@@ -80,38 +85,39 @@ do
   imgName=${reqImgURL##*/}
 
   # Create Path Dir.
-  mkdir -p $path
+  mkdir -p "$path"
```
Comment (whizzzkid) [inline bingwallpaper:88]: not needed

File: bingwallpaper [hunk 9]
```diff
@@ -80,38 +85,39 @@ do
...
 
   # Saving Image to collection.
-  curl -s -o $path$imgName $reqImgURL
+  curl -s -o "$path$imgName" "$reqImgURL"
```
Comment (whizzzkid) [inline bingwallpaper:91]: ditto

File: bingwallpaper [hunk 10]
```diff
@@ -80,38 +85,39 @@ do
...
 
   # Logging.
   echo "Saving image to $path$imgName"
 
-  if [ "$XDG_CURRENT_DESKTOP" = "XFCE" ]
+  : "${XDG_CURRENT_DESKTOP:=}"
```
Comment (whizzzkid) [inline bingwallpaper:96]: this is confusing.

File: debian/changelog [hunk 11]
```diff
@@ -1,11 +0,0 @@
-bingwallpaper (0.5-1) UNRELEASED; urgency=medium
```
Comment (whizzzkid) [inline debian/changelog:1]: please do not remove existing files.

File: debian/control [hunk 12]
```diff
@@ -1,12 +0,0 @@
-Source: bingwallpaper
```
Comment (whizzzkid) [inline debian/control:1]: do not remove existing files

File: debian/copyright [hunk 13]
```diff
@@ -1,21 +0,0 @@
-Format: http://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
```
Comment (whizzzkid) [inline debian/copyright:1]: do not remove existing files

File: debian/install [hunk 14]
```diff
@@ -1 +0,0 @@
-bingwallpaper usr/bin
```
Comment (whizzzkid) [inline debian/install:1]: do not remove existing files

File: debian/postinst [hunk 15]
```diff
@@ -1,25 +0,0 @@
-#!/bin/sh
```
Comment (whizzzkid) [inline debian/postinst:1]: do not remove existing files

File: debian/rules [hunk 16]
```diff
@@ -1,12 +0,0 @@
-#!/usr/bin/make -f
```
Comment (whizzzkid) [inline debian/rules:1]: do not remove existing files

File: debian/source/format [hunk 17]
```diff
@@ -1 +0,0 @@
-1.0
```
Comment (whizzzkid) [inline debian/source/format:1]: do not remove existing files

**Review by whizzzkid — CHANGES_REQUESTED** [review]
Other files in there exist for a purpose. launchpad automatically pulls the repo and builds this into a debian package which is then pushed over apt as an update. If you remove that then everything fails.

Please checkout all the existing files one by one back into this commit except `bingwallpaper` Then make the changes requested in that file only.

Rest this looks better. Looking forward to review this.

**Comment by aggsol** [comment]
Aaargh sorry, I just wanted the first commit as PR. The rest are my own changes. :-(

**Pull request closed without merging**